eva/io_descriptors/eva_arguments.py

```diff
@@ -0,0 +1,140 @@
+
+
+class IOArgument(object):
+    """
+    Base class for the data types that are used inside Eva. This class is inherited by the NumPyArray,
+    PyTorchTensor classes.
+    The functions are implemented in the child classes.
+
+    """
+
+    @abstractmethod
+    def __init__(self) -> None:
+        pass
+
+    @abstractmethod
+    def check_type(self, input_object: any) -> bool:
+        """Checks the type of the input_object with
+
+        Args:
+            input_object (any): the object whose type should be checked. The required type is given in the constructor.
+
+        Returns:
+            bool: True if the type of input_object matches the required type. False otherwise.
+
+        """
+        pass
+
+    @abstractmethod
+    def check_shape(self, input_object: any, required_shape: tuple = None) -> bool:
+        """Checks the shape of the input_object with
+
+        Args:
+            input_object (any): the object whose shape should be checked. The required shape is given in the constructor.
+
+        Returns:
+            bool: True if the shape of input_object matches the required type. False otherwise.
+
+        """
+
+        pass
+
+    @abstractmethod
+    def arg_name(self):
+        """Returns the name of the EvaArgument.
+
+        It is used in the construction of the error messages.
+        """
+        pass
+
+    @abstractmethod
+    def is_output_columns_set(self):
+        """Checks if the output columns are set.
+
+        This is used for EvaArguments which are of PandasDataFrame type.
+
+        """
+        pass
+
+    @abstractmethod
+    def check_column_names(self, output_object):
+        """Checks if the output column names match the required column names list.
+
+        Args:
+            output_object (any): the object whose columns should be checked. It should be of type PandasDataFrame.
+                        The required column list is given in the constructor.
+
+        Returns:
+            bool: True if the column names of output_object matches the required columns list. False otherwise.
+
+        """
+        pass
+
+    @abstractmethod
+    def reshape(self, input_object: any):
+        """function to reshape the input object to the required shape given in the constructor
+
+        Args:
+            input_object (any): the object which must be reshaped
+        """
+        pass
+
+    @abstractmethod
+    def convert_data_type(self, input_object: any):
+        """convert the data type of input object to that given in the constructor.
+
+        Args:
+            input_object (any): object whose data type must be updated.
+        """
+
+    def is_shape_defined(self):
+        """returns True if the size has been specified. False otherwise"""
+        return True
+
+    def is_dtype_defined(self):
+        """returns True if the dtype has been specified. False otherwise"""
+        return True
+
+
+class IOColumnArgument(object):
+    """ """
```

Comment:
Add a docstring here explaining what this would be sued for, essentially for defining types pertaining to an individual column/field.